### DISCUSSION TITLE
postgresql: use identity columns instead of “serial”

### DISCUSSION THREAD

**Discussion by Sam-Apostel** [payloadcms/payload#8096] [Feature Requests & Ideas]
I am running into "Internal error: error: type "serial" does not exist" when setting up payload 3 beta 99 with a posgresql connection to xata.io

Digging through several issues I discovered that the 'serial' type in postgresql is outdated and "should be deprecated". 

> mentioned in drizzle-kit@0.23.0 release notes https://github.com/drizzle-team/drizzle-orm/releases/tag/0.32.0

> note about this on the postgresql wiki: https://wiki.postgresql.org/wiki/Don%27t_Do_This#Don.27t_use_serial

Should the `serial` type be replaced with an `Identity column`? 
Will that change help with improving compatibility with more serverless postgresql providers?

**Comment by DanRibbens** [comment]
I've discussed this with the drizzle team. It is something we looked at but haven't prioritized. A workaround right now for you is to set the `idType: 'uuid'` which wiill replace all `serial` columns out for `uuid`.

If you're interested in adding the `idType: 'identityColumn'` as a PR, we'd appreciate the extra hand!

**Reply by Sam-Apostel** [reply]
the "payload_preferences_rels" table create sql does not seam to adhere to `idType: 'uuid'`. I'll try to PR later if I have some more time to figure out the codebase. I'm new to payload so it will take me a bit longer :D

Thanks for the tips

**Reply by notflip** [reply]
Is this something that's been looked at? I keep getting this error as well, it's too late to switch to UUID.

**Reply by Jeandcc** [reply]
+1 on those `_rels` tables seemingly not adhering to idType: 'uuid'
